src/Package/Impl/ProjectSystem/PropertyPages/PropertyPageControl.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using System.Windows.Controls;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem.PropertyPages {
+    internal class PropertyPageControl : UserControl {
+        private bool _isDirty;
+        private bool _ignoreEvents;
+
+        public event EventHandler StatusChanged;
+
+        public PropertyPageViewModel ViewModel {
+            get { return DataContext as PropertyPageViewModel; }
+            set { DataContext = value; }
+        }
+
+        public bool IsDirty {
+            get { return _isDirty; }
+            set {
+                // Only process real changes
+                if (value != _isDirty && !_ignoreEvents) {
+                    _isDirty = value;
+                    OnStatusChanged(new EventArgs());
+                }
+            }
+        }
+
+        public virtual void InitializePropertyPage(PropertyPageViewModel viewModel) {
+            _ignoreEvents = true;
+            IsDirty = false;
+            ViewModel = viewModel;
+            ViewModel.PropertyChanged += ViewModel_PropertyChanged;
```

Comment:
It's all about propagating 'dirty' bit flag from view model to control then to page.  Page calls IPropertyPageSite.OnStatusChange which updates the UI with a '*' and causes an Apply on the page.

I find it all very confusing, but it works.  I took it as-is from the dotnet project system.

I can try to simplify it if you think it's worth it.